app/react/browser/components/Paginator.react.js

```diff
@@ -0,0 +1,37 @@
+import React, { PropTypes as RPT, PureComponent } from 'react';
+import ReactPaginate from 'react-paginate';
+import { withRouter } from 'react-router';
+
+@withRouter
+export default class Paginator extends PureComponent {
+  static defaultProps = {
+    totalPages: 1,
+  };
+  static propTypes = {
+    location: RPT.shape({ query: { page: RPT.string } }).isRequired,
+    router: RPT.shape({ push: RPT.func }).isRequired,
+    totalPages: RPT.number,
+  };
+
+  handleChangePage = ({ selected: page }) => {
+    this.props.router.push({
```

Comment:
It'll be more consistent, from app architecture point of view, to use of action `push` from `react-router-redux`.